src/transformers/models/pop2piano/tokenization_pop2piano.py

```diff
@@ -0,0 +1,428 @@
+        return midi, notes
+
+    # Taken from the original code
+    # Please see https://github.com/sweetcocoa/pop2piano/blob/fac11e8dcfc73487513f4588e8d0c22a22f2fdc5/midi_tokenizer.py#L257
+    def relative_tokens_to_notes(self, tokens, start_idx, cutoff_time_idx=None):
+        if tokens[0] >= (
+            self.vocab_size_special + self.vocab_size_note + self.vocab_size_velocity + self.vocab_size_time
+        ):
+            tokens = tokens[1:]
+
+        words = [self._convert_id_to_token(token, time_idx_offset=0) for token in tokens]
+
+        if hasattr(start_idx, "item"):
+            """if numpy or torch tensor"""
+            start_idx = start_idx.item()
+
+        current_idx = start_idx
+        current_velocity = 0
+        note_onsets_ready = [None for i in range(self.vocab_size_note + 1)]
+        notes = []
+        for type, number in words:
+            if type == self.encoder["TOKEN_SPECIAL"]:
+                if number == 1:
+                    break
+            elif type == self.encoder["TOKEN_TIME"]:
+                current_idx += number
+                if cutoff_time_idx is not None:
+                    current_idx = min(current_idx, cutoff_time_idx)
+
+            elif type == self.encoder["TOKEN_VELOCITY"]:
+                current_velocity = number
+            elif type == self.encoder["TOKEN_NOTE"]:
+                pitch = number
+                if current_velocity == 0:
+                    # note_offset
+                    if note_onsets_ready[pitch] is None:
+                        # offset without onset
+                        pass
+                    else:
+                        onset_idx = note_onsets_ready[pitch]
+                        if onset_idx >= current_idx:
+                            # No time shift after previous note_on
+                            pass
+                        else:
+                            offset_idx = current_idx
+                            notes.append([onset_idx, offset_idx, pitch, self.encoder["DEFAULT_VELOCITY"]])
+                            note_onsets_ready[pitch] = None
+                else:
+                    # note_on
+                    if note_onsets_ready[pitch] is None:
+                        note_onsets_ready[pitch] = current_idx
+                    else:
+                        # note-on already exists
+                        onset_idx = note_onsets_ready[pitch]
+                        if onset_idx >= current_idx:
+                            # No time shift after previous note_on
+                            pass
+                        else:
+                            offset_idx = current_idx
+                            notes.append([onset_idx, offset_idx, pitch, self.encoder["DEFAULT_VELOCITY"]])
+                            note_onsets_ready[pitch] = current_idx
+            else:
+                raise ValueError("Token type not understood!")
+
+        for pitch, note_on in enumerate(note_onsets_ready):
```

Comment:
I'm guessing this represents `notes_onset`? Let's use the full name to make the code easier to understand 

```suggestion
        for pitch, note_onset in enumerate(note_onsets_ready):
```